### PR TITLE
Worked on the Secret Fields in ingestion form

### PR DESCRIPTION
**Pull request by Ankit-Keshari-Vituity**
## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-web-react/src/app/ingest/source/builder/RecipeForm/SecretField/SecretField.tsx [hunk 1]
```diff
@@ -36,25 +36,30 @@ interface SecretFieldProps {
...
-                )}
-            >
-                {secrets.map((secret) => (
-                    <Select.Option key={secret.urn} value={`\${${secret.name}}`}>
-                        {secret.name}
-                    </Select.Option>
-                ))}
-            </Select>
-        </StyledFormItem>
+        <>
```
Comment (chriscollins3456) [inline datahub-web-react/src/app/ingest/source/builder/RecipeForm/SecretField/SecretField.tsx:39]: why add the react fragments here? I think you can remove!

File: datahub-web-react/src/app/ingest/source/builder/RecipeForm/SecretField/SecretField.tsx [hunk 2]
```diff
@@ -36,25 +36,30 @@ interface SecretFieldProps {
...
+            <StyledFormItem name={field.name} label={field.label} tooltip={field.tooltip} removeMargin={!!removeMargin}>
+                <Tooltip title="This field requires you to use a DataHub Secret. For more information on Secrets in DataHub, please review the docs.">
```
Comment (chriscollins3456) [inline datahub-web-react/src/app/ingest/source/builder/RecipeForm/SecretField/SecretField.tsx:41]: can you make "the docs" a link to https://datahubproject.io/docs/ui-ingestion/#creating-a-secret

**Review by chriscollins3456 — COMMENTED** [review]
looking good! just two small comments then this will be good to ship

File: datahub-web-react/src/app/ingest/source/builder/RecipeForm/SecretField/SecretField.tsx [hunk 3]
```diff
@@ -34,32 +34,48 @@ interface SecretFieldProps {
...
+            </p>
+        </div>
+    );
+}
+
+function SecretField({ field, secrets, removeMargin, refetchSecrets }: SecretFieldProps) {
+    return (
+        <StyledFormItem name={field.name} label={field.label} tooltip={field.tooltip} removeMargin={!!removeMargin}>
+            <Tooltip title={getTooltip}>
+                <span style={{ color: ANTD_GRAY[7] }}>Secret Field</span>
```
Comment (chriscollins3456) [inline datahub-web-react/src/app/ingest/source/builder/RecipeForm/SecretField/SecretField.tsx:59]: can we make this a styled component instead of inline?

Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/ingest/source/builder/RecipeForm/SecretField/SecretField.tsx]: Fixed and Commited!

File: datahub-web-react/src/app/ingest/source/builder/RecipeForm/SecretField/SecretField.tsx (hunk 2 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/ingest/source/builder/RecipeForm/SecretField/SecretField.tsx]: Fixed and Committed!

File: datahub-web-react/src/app/ingest/source/builder/RecipeForm/SecretField/SecretField.tsx (hunk 1 above)
Comment (Ankit-Keshari-Vituity) [reply datahub-web-react/src/app/ingest/source/builder/RecipeForm/SecretField/SecretField.tsx]: Fixed and Commited!

File: datahub-web-react/src/app/ingest/source/builder/RecipeForm/SecretField/SecretField.tsx [hunk 4]
```diff
@@ -1,15 +1,24 @@
...
+import { ANTD_GRAY } from '../../../../../entity/shared/constants';
 
 const StyledDivider = styled(Divider)`
     margin: 0;
 `;
 
-export const StyledFormItem = styled(Form.Item)<{ alignLeft?: boolean; removeMargin: boolean }>`
+export const SecretFieldLabel = styled.span`
+    color: ${ANTD_GRAY[7]};
+`;
```
Comment (chriscollins3456) [inline datahub-web-react/src/app/ingest/source/builder/RecipeForm/SecretField/SecretField.tsx:15]: can delete now that you're not using it

File: datahub-web-react/src/app/ingest/source/builder/RecipeForm/SecretField/SecretField.tsx [hunk 5]
```diff
@@ -34,9 +58,36 @@ interface SecretFieldProps {
     refetchSecrets: () => void;
 }
 
+function SecretFieldTooltip({ tooltipLabel }: { tooltipLabel?: string | ReactNode }) {
+    return (
+        <div>
+            {tooltipLabel}
+            <hr />
```
Comment (chriscollins3456) [inline datahub-web-react/src/app/ingest/source/builder/RecipeForm/SecretField/SecretField.tsx:65]: can you add a check that the tooltipLabel exists first before adding this horizontal line? so you could do:
```
{tooltipLabel && (<>{tooltipLabel}<hr /></>)}
```

**Review by chriscollins3456 — APPROVED** [review]

**Pull request merged**